apps/checkout-app/backend/payments/providers/mollie/webhookHandler.ts

```diff
@@ -2,40 +2,131 @@ import { OrderStatus } from "@mollie/api-client";
 
 import { TransactionCreateMutationVariables } from "@/checkout-app/graphql";
 
-import { getMollieClient } from "./utils";
+import { getMollieEventName, getMollieClient } from "./utils";
+import { getTransactionAmount } from "../../utils";
+
+export const MOLLIE_PAYMENT_PREFIX = "mollie";
 
 export const verifyPayment = async (
   id: string
 ): Promise<TransactionCreateMutationVariables | undefined> => {
   const mollieClient = await getMollieClient();
 
-  const { status, amountCaptured, metadata, method, amount } =
+  const { status, amountCaptured, amountRefunded, metadata, method, amount } =
     await mollieClient.orders.get(id);
 
+  const type = `${MOLLIE_PAYMENT_PREFIX}-${method}`;
+  const reference = id;
+  const eventName = getMollieEventName(status);
+
+  const getAmount = getTransactionAmount({
+    authorized: amount?.value,
+    voided: undefined,
+    refunded: amountRefunded?.value,
+    charged: amountCaptured?.value,
+  });
+
+  if (status === OrderStatus.created || status === OrderStatus.pending) {
+    return {
```

Comment:
Can it be divided into smaller functions?

Reply:
I mean... not really? We could perhaps refactor each status from Mollie into a separate function, but imho that would be an overkill. This function is not that long and most of the space is taken from the returned object